src/praisonai-bot/praisonai_bot/bots/_metrics.py

```
@@ -77,6 +94,27 @@ def _escape(value: str) -> str:
+    """Render a bucket upper bound without a trailing ``.0`` for whole numbers."""
+    if bound == int(bound):
+        return str(int(bound))
+    return repr(bound)
+
+
+class _HistData:
+    """Mutable per-label-set histogram accumulator."""
+
+    __slots__ = ("counts", "sum", "count")
+
+    def __init__(self, counts: List[int], sum: float, count: int) -> None:
```

Ruff (0.16.1)

[error] Function argument `sum` is shadowing a Python builtin

(A002)